### PR TITLE
Add support for TaxId resource and APIs

### PR DESCRIPTION
**Pull request by remi-stripe**
cc @stripe/api-libraries @nicku-stripe

### CODE REVIEW

**Comment by remi-stripe** [comment]
This is now ready to be reviewed.
r? @ob-stripe

File: lib/Customer.php [hunk 1]
```diff
@@ -202,4 +203,54 @@ public static function allSources($id, $params = null, $opts = null)
...
     }
+
+    /**
+     * @param string|null $id The ID of the customer on which to create the tax id.
+     * @param array|null $params
+     * @param array|string|null $opts
+     *
+     * @return ApiResource
+     */
+    public static function createTaxId($id, $params = null, $opts = null)
```
Comment (remi-stripe) [inline lib/Customer.php]: Flagging that `tax_ids` the collection is returned on the Customer object, like sources and subscriptions. Not sure anymore if that changes things

File: lib/TaxId.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    /**
+     * @param array|string $_id
+     * @param array|string|null $_opts
+     *
+     * @throws \Stripe\Error\InvalidRequest
+     */
+    public static function retrieve($_id, $_opts = null)
+    {
+        $msg = "Tax Ids cannot be accessed without a customer ID. " .
+               "Retrieve a Tax Id using \$account->retrieveTaxId('tax_id') instead.";
```
Comment (ob-stripe) [inline lib/TaxId.php]:
```suggestion
               "Retrieve a Tax Id using Customer::retrieveTaxId('tax_id') instead.";
```

File: tests/Stripe/CustomerTest.php [hunk 3]
```diff
@@ -6,6 +6,7 @@ class CustomerTest extends TestCase
 {
     const TEST_RESOURCE_ID = 'cus_123';
     const TEST_SOURCE_ID = 'ba_123';
+    const TEST_TAX_ID = 'txi_123';
```
Comment (ob-stripe) [inline tests/Stripe/CustomerTest.php]: Should maybe be `TEST_TAX_ID_ID` since `tax_id` is the name of the resource, but that's silly :p

**Comment by remi-stripe** [comment]
@ob-stripe PTAL

**Pull request merged**